kong/plugins/moesif/regex_config_helpers.lua

```diff
@@ -0,0 +1,180 @@
+local _M = {}
+
+-- Function to perform the regex matching with event value and condition value
+-- @param  `event_value`     Value associated with event (request)
+-- @param  `condition_value` Value associated with the regex config condition
+-- @param  `regex_matched`   Boolean flag with regex matched from previous regex config condition
+-- @return `regex_matched`   Boolean flag to determine if the regex match was successful 
+local function regex_match (event_value, condition_value, regex_matched)
+    -- Perform regex match between event value and regex config condition value
+    if string.match(event_value, condition_value) then 
+        if regex_matched == nil then 
+            regex_matched = true
+        else
+            regex_matched = (regex_matched and true) 
+        end 
+    else
+        if regex_matched == nil then 
+            regex_matched = false
+        else
+            regex_matched = (regex_matched and false)
+        end 
+    end
+    return regex_matched
+end
+
+-- Function to fetch the sample rate and determine if request needs to be block or not
+-- @param  `gr_regex_configs`        Regex configs associated with the governance rule
+-- @param  `request_config_mapping`  Config associated with the request
+-- @return `sample_rate, block`      Sample rate and boolean flag (block or not)
+function _M.fetch_sample_rate_block_request_on_regex_match(gr_regex_configs, request_config_mapping)
+    -- Iterate through the list of governance rule regex configs
+    for _, regex_rule in pairs(gr_regex_configs) do
+        -- Fetch the sample rate
+        local sample_rate = regex_rule["sample_rate"]
+        -- Fetch the conditions
+        local conditions = regex_rule["conditions"]
+        -- Bool flag to determine if the regex conditions are matched
+        local regex_matched = nil 
+        -- Create a table to hold the conditions mapping (path and value)
+        local condition_table = {}
+
+        -- Iterate through the regex rule conditions and map the path and value
+        for _, condition in pairs(conditions) do
+            -- Check if the path does not exists already in the table (when multiple path like request.route exist in the conditions) 
+            if condition_table[condition["path"]] ~= nil then 
+                -- Create a table to hold the existing values
+                local append_value = {}
```

Comment:
this is array instead of table?

Reply:
I am bit worried of using creating local table/array every time. This can be avoided, if in line 64, if you create a table to startwith, and this logic here will be simple..Just to add to the array?